src/constraint/unique_constraint.c

```diff
@@ -138,7 +142,9 @@ static bool Constraint_EnforceUniqueEntity
 	if(iter != NULL) {
 		RediSearch_ResultsIteratorFree(iter);
 	} else {
-		RediSearch_QueryNodeFree(root);
+		for(uint8_t i = 0; i < _c->n_attr; i++) {
```

Comment:
Keep the original `i` from the query construction loop above. only free nodes in the range `0..I`
this removes the need for `memset` and avoids calling `RediSearch_QueryNodeFree` on redundant entries.